pyproject.toml

```diff
@@ -20,7 +20,7 @@ classifiers = [
     "Programming Language :: Python :: 3.10",
 ]
 dynamic = ["version"]
-dependencies = ['psygnal', 'pydantic', 'in-n-out', 'typing_extensions']
+dependencies = ['psygnal', 'pydantic', 'in-n-out>=0.1.5', 'typing_extensions']
```

Comment:
I do not see `typing_extensions` usage in this PR. Is this not added earlier, or is some artifact from the development phase?

Reply:
yeah, it's not added or new in this PR,  the change in this line is to `in-n-ou`

Reply:
Ahh. But still, typing_extenstions should be in version at least 3.10?

Reply:
Well I import from it without checking python version, so yes. 

Typing extensions itself has all the version guard conditionals, so if the typing module offers the thing you're using, it will just use it.  So there's no need to only install it conditionally

Reply:
so `ParamSpec` is present in all version of `typing_extensions`? I think that 3.10.x is a minimal version of `typing_extensions` to have code working. Or I'm wrong?

Reply:
Oh I see what you're saying, no it's not, but as you'll see in the code usage, I specifically try to import param spec, and then fall back to using type var.  the reason here is that an old version of tensorflow very annoyingly pinned typing extensions, and so if I pin a min value here, it's incompatible with tensorflow (and we'll get an error here if someone installs TF on top).  

See https://github.com/napari/superqt/pull/101
And https://github.com/napari/app-model/pull/49

Reply:
gonna resolve this topic to keep the visual clutter down.  but let me know if there's anything confusing about why/what i did there